pioinstaller/penv.py

```diff
@@ -86,6 +135,20 @@ def create_virtualenv_with_download(python_exe, penv_dir):
     command = [python_exe, venv_path, penv_dir]
     log.debug("Creating virtual environment: %s", " ".join(command))
     subprocess.check_output(command)
+    add_state_info(python_exe, penv_dir)
+    return penv_dir
+
+
+def create_virtualenv_with_portable_python(penv_dir):
+    python_exe = download_portable_python()
+    clean_dir(penv_dir)
+    venv_path = download_virtualenv_script(core.get_cache_dir())
+    if not venv_path:
+        raise exception.PIOInstallerException("Could not find virtualenv script")
+    command = [python_exe, venv_path, penv_dir]
+    log.debug("Creating virtual environment: %s", " ".join(command))
+    subprocess.check_output(command)
+    add_state_info(python_exe, penv_dir)
```

Comment:
Why you don't re-use `create_virtualenv_with_local` and then with `create_virtualenv_with_download`? No need to duplicate code.